test/test_gridfs.py

```diff
@@ -41,35 +42,66 @@
 from pymongo.synchronous.database import Database
 from pymongo.synchronous.mongo_client import MongoClient
 
-
-class JustWrite(threading.Thread):
-    def __init__(self, fs, n):
-        threading.Thread.__init__(self)
-        self.fs = fs
-        self.n = n
-        self.daemon = True
-
-    def run(self):
-        for _ in range(self.n):
-            file = self.fs.new_file(filename="test")
-            file.write(b"hello")
-            file.close()
-
-
-class JustRead(threading.Thread):
-    def __init__(self, fs, n, results):
-        threading.Thread.__init__(self)
-        self.fs = fs
-        self.n = n
-        self.results = results
-        self.daemon = True
-
-    def run(self):
-        for _ in range(self.n):
-            file = self.fs.get("test")
-            data = file.read()
-            self.results.append(data)
-            assert data == b"hello"
+_IS_SYNC = True
+
+if _IS_SYNC:
+
+    class JustWrite(threading.Thread):
```

Comment:
Can we use the shared thread/task api that Noah introduced here?

Reply:
yes! Just changed that!